base/c.jl

```diff
@@ -23,7 +23,7 @@ if is_windows()
 else
     typealias Clong Int
     typealias Culong UInt
-    typealias Cwchar_t Int32
+    typealias Cwchar_t UInt32
```

Comment:
> should I reverse it to `Int32`

Yes.

> Aren't the used to represent characters and therefore negatives are not useful?

Blame whoever define them to be signed in C. `Cchar` is also signed on linux afaik.